core/src/main/java/org/mule/runtime/core/policy/CompositeOperationPolicy.java

```diff
@@ -0,0 +1,96 @@
+import org.mule.runtime.core.api.policy.OperationPolicyParametersTransformer;
+import org.mule.runtime.core.api.policy.SourcePolicyParametersTransformer;
+import org.mule.runtime.core.api.processor.Processor;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * {@link OperationPolicy} created from a list of {@link ParameterizedPolicy}.
+ * <p>
+ * Takes care of chaining the list of {@link ParameterizedPolicy} to create a single
+ * policy that can be applied to a source.
+ *
+ * @since 4.0
+ */
+public class CompositeOperationPolicy implements OperationPolicy {
+
+  private List<ParameterizedPolicy> parameterizedPolicies;
+  private Optional<OperationPolicyParametersTransformer> operationPolicyParametersTransformer;
+  private PolicyStateHandler policyStateHandler;
+
+  /**
+   * Creates a new composite policy.
+   *
+   * @param parameterizedPolicies list of {@link ParameterizedPolicy} to chain together.
+   * @param operationPolicyParametersTransformer transformer from the operation parameters to a message and vice versa.
+   * @param policyStateHandler state handler for policies execution.
+   */
+  public CompositeOperationPolicy(List<ParameterizedPolicy> parameterizedPolicies,
+                                  Optional<OperationPolicyParametersTransformer> operationPolicyParametersTransformer,
+                                  PolicyStateHandler policyStateHandler) {
+    this.parameterizedPolicies = parameterizedPolicies;
+    this.operationPolicyParametersTransformer = operationPolicyParametersTransformer;
+    this.policyStateHandler = policyStateHandler;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Event process(Event operationEvent, Processor nextProcessor, OperationParametersProcessor operationParametersProcessor)
+      throws Exception {
+    return new NextOperationCall(operationEvent, nextProcessor, operationParametersProcessor).process(operationEvent);
+  }
+
+  /**
+   * Inner class that implements the actually chaining of policies.
+   */
+  public class NextOperationCall implements Processor {
+
+    private int index = 0;
+    private Processor operationProcessor;
+    private Event operationEvent;
+    private OperationParametersProcessor operationParametersProcessor;
+    private Event operationResponseEvent;
+
+    public NextOperationCall(Event operationEvent, Processor operationProcessor,
+                             OperationParametersProcessor operationParametersProcessor) {
+      this.operationEvent = operationEvent;
+      this.operationProcessor = operationProcessor;
+      this.operationParametersProcessor = operationParametersProcessor;
+    }
+
+    @Override
+    public Event process(Event event) throws MuleException {
+      if (index >= parameterizedPolicies.size()) {
+        operationResponseEvent = operationProcessor.process(event);
+        return operationResponseEvent;
+      }
+      ParameterizedPolicy parameterizedPolicy = parameterizedPolicies.get(index);
```

Comment:
I added the explanation as javadoc.
/**
     * When this policy is processed, it will use the {@link NextOperationCall} which will keep track of the different policies to be applied and
     * the current index of the policy under execution.
     * <p>
     * The first time, the first policy in the {@code parameterizedPolicies} will be executed, it will receive as it next operation the same
     * instance of {@link NextOperationCall}, and since {@link NextOperationCall} keeps track of the policy executed, it will execute the following
     * policy in the chain until the finaly policy it's executed in which case then next operation of it, it will be the operation execution.
     */